mod-source-record-storage-server/src/main/java/org/folio/rest/impl/SourceStorageBatchImpl.java

```diff
@@ -101,4 +102,30 @@ public void putSourceStorageBatchParsedRecords(RecordCollection entity, Map<Stri
       }
     });
   }
+
+  @Override
+  public void postSourceStorageBatchParsedRecordsFetch(FetchParsedRecordsBatchRequest entity,
+                                                       Map<String, String> okapiHeaders,
+                                                       Handler<AsyncResult<Response>> asyncResultHandler,
+                                                       Context vertxContext) {
+    vertxContext.runOnContext(v -> {
+      try {
+        recordService.fetchParsedRecords(entity, tenantId)
+          .map(parsedRecordsBatchResponse -> {
+            if (!parsedRecordsBatchResponse.getRecords().isEmpty()) {
+              return PostSourceStorageBatchParsedRecordsFetchResponse.respond200WithApplicationJson(parsedRecordsBatchResponse);
+            } else {
+              LOG.warn("postSourceStorageBatchParsedRecordsFetch:: Batch of parsed records was fetched, but no records were found");
+              return PostSourceStorageBatchParsedRecordsFetchResponse.respond500WithApplicationJson(parsedRecordsBatchResponse);
```

Comment:
I think it could be 400 or 404. 500 is for unexpected cases.

Reply:
In most cases SRS returns 500 status with empty list of records.
But I believe it's better to return 200 with empty list